File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 2/11/2026, 11:08:55 AM
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;

        for (int i =0; i< size(nums); ++i){
            if(mp.find(target-nums[i])!= mp.end()){
                return {i, mp[target-nums[i]]};
            }
            mp[nums[i]] = i;
        }
        return {};
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 2/11/2026, 11:08:56 AM
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        mp = {}

        for i,num in enumerate(nums):
            if target-num in mp:
                return [mp[target-num],i]
            mp[num] = i


        